### PR TITLE
feat: show usage count when delete variable

### PR DESCRIPTION
**Pull request by TrySound**
Ref https://github.com/webstudio-is/webstudio/issues/3080

Now user can see how many usages left. It can give some hint where to look. We will explore in the future how to show the list of usages.

Also fixed usage detection on pages.

Screenshot 2024-06-01 at 18 45 53

### CODE REVIEW

File: apps/builder/app/builder/features/settings-panel/variables-section.tsx [hunk 1]
```diff
@@ -124,6 +126,20 @@ const $usedVariables = computed(
         }
       }
     }
+    if (page) {
```
Comment (kof) [inline apps/builder/app/builder/features/settings-panel/variables-section.tsx:129]: that list looks extremely suspicious: how are we supposed to know we need to update it if we add more things to page settings?

**Review by kof — APPROVED** [review]
- Great first step in this story
- I really don't see that logic with a hardcoded list of settings getting merged without us having issues later

File: apps/builder/app/builder/features/settings-panel/variables-section.tsx (hunk 1 above)
Comment (kof) [reply apps/builder/app/builder/features/settings-panel/variables-section.tsx:129]: Please don't merge it like this.

Comment (TrySound) [reply apps/builder/app/builder/features/settings-panel/variables-section.tsx:129]: no way, we have it in pages ui, in compiler, in utils and in variables.

Comment (TrySound) [reply apps/builder/app/builder/features/settings-panel/variables-section.tsx:129]: Will try to generalize with incremental updates in the future

**Pull request merged**